x-pack/platform/plugins/shared/data_connectors/server/routes/index.ts

```diff
@@ -0,0 +1,344 @@
+      },
+      security: {
+        authz: {
+          enabled: false,
+          reason: 'This route is opted out from authorization',
+        },
+      },
+    },
+    async (context, request, response) => {
+      const coreContext = await context.core;
+
+      try {
+        const { name, type, token } = request.body;
+        const [, { actions, dataSourcesRegistry }] = await getStartServices();
+
+        const dataCatalog = dataSourcesRegistry.getCatalog();
+        const dataConnectorTypeDef = dataCatalog.get(type);
+        if (!dataConnectorTypeDef) {
+          return response.customError({
+            statusCode: 400,
+            body: {
+              message: `Data connector type "${request.body.type}" not found`,
+            },
+          });
+        }
+
+        const connectorType = dataConnectorTypeDef.stackConnector.type;
+        const secrets = buildSecretsFromConnectorSpec(connectorType, token);
+
+        const actionsClient = await actions.getActionsClientWithRequest(request);
+        const stackConnector = await actionsClient.create({
+          action: {
+            name: `${type} stack connector for data connector '${name}'`,
+            actionTypeId: connectorType,
+            config: {},
+            secrets,
+          },
+        });
+
+        const savedObjectsClient = coreContext.savedObjects.client;
+        const now = new Date().toISOString();
+        const savedObject = await savedObjectsClient.create(DATA_CONNECTOR_SAVED_OBJECT_TYPE, {
+          name,
+          type,
+          config: {},
+          features: [],
+          createdAt: now,
+          updatedAt: now,
+          workflowIds: [],
+          toolIds: [],
+          kscId: stackConnector.id,
+        });
+
+        return response.ok({
+          body: {
+            message: 'Data connector created successfully!',
+            dataConnectorId: savedObject.id,
+            stackConnectorId: stackConnector.id,
+          },
+        });
+      } catch (error) {
+        logger.error(`Error creating data connector: ${(error as Error).message}`);
+        return response.customError({
+          statusCode: 500,
+          body: {
+            message: `Failed to create data connector: ${(error as Error).message}`,
+          },
+        });
+      }
+    }
+  );
+
+  // Delete all data connectors
+  router.delete(
+    {
+      path: '/api/data_connectors',
+      validate: false,
+      security: {
+        authz: {
+          enabled: false,
+          reason: 'This route is opted out from authorization',
+        },
+      },
+    },
+    async (context, request, response) => {
+      const coreContext = await context.core;
+
+      try {
+        const savedObjectsClient = coreContext.savedObjects.client;
+        const findResponse: SavedObjectsFindResponse<DataConnectorAttributes> =
+          await savedObjectsClient.find({
+            type: DATA_CONNECTOR_SAVED_OBJECT_TYPE,
+            perPage: 1000,
```

Comment:
This doesn't look to me like it loops, so it would just delete the first 1000? Which is _probably_ all of them, but why not just loop to be sure?

Reply:
Created https://github.com/elastic/search-team/issues/12118 to track - I reckon once `DELETE` actually deletes workflows & tools too, this can be picked up.